addons/website_sale/controllers/main.py

```diff
@@ -426,6 +430,10 @@ def shop(self, page=0, category=None, search='', min_price=0.0, max_price=0.0, p
 
     @route(['/shop/<model("product.template"):product>'], type='http', auth="public", website=True, sitemap=True)
     def product(self, product, category='', search='', **kwargs):
+
+        not_allowed = self.check_shop_access()
+        if not_allowed:
+            return not_allowed
```

Comment:
Maybe it would be more understandable what is returned.
```suggestion
        redirection = self.check_shop_access()
        if redirection:
            return redirection
```